JNetHack 5.0.0-0.1 is a Windows-oriented test release: only sys/windows/Makefile.nmake
knows about japanese/, so the Unix build has nothing to link jlib.c and jconj.c into,
POSIX_ICONV/ICUTF8 are never defined, and the one call site of version_string_j() that
was left behind still uses the old signature.  Restore what 3.6.7-0.2 had.

Submitted upstream as https://github.com/jnethack/jnethack-alpha/pull/30

--- a/include/extern.h
+++ b/include/extern.h
@@ -3568,6 +3568,9 @@ extern void vault_gd_watching(unsigned int);
 /* ### version.c ### */
 
 extern char *version_string(char *, size_t bufsz) NONNULL NONNULLARG1;
+#if 1 /*JP*/
+extern char *version_string_j(char *, size_t) NONNULL NONNULLARG1;
+#endif
 extern char *getversionstring(char *, size_t bufsz) NONNULL NONNULLARG1;
 extern char *status_version(char *, size_t, boolean) NONNULL NONNULLARG1;
 extern int doversion(void);
--- a/include/unixconf.h
+++ b/include/unixconf.h
@@ -64,6 +64,10 @@
                      * For example, platforms using the GNU libraries,
                      * Linux, Solaris 2.x
                      */
+#if 1 /*JP*/
+#define POSIX_ICONV /* use POSIX iconv functions */
+#define ICUTF8      /* ソースがUTF-8なので内部コードもUTF-8 */
+#endif
 
 /* #define RANDOM */ /* if neither random/srandom nor lrand48/srand48
                         is available from your system */
--- a/src/end.c
+++ b/src/end.c
@@ -674,7 +674,7 @@ dump_everything(
        but we only have access to the one it finished under */
     putstr(0, 0, getversionstring(pbuf, sizeof pbuf));
 #if 1 /*JP*/
-    putstr(0, 0, version_string_j(pbuf));
+    putstr(0, 0, version_string_j(pbuf, sizeof pbuf));
 #endif
     putstr(0, 0, "");
 
--- a/sys/unix/Makefile.src
+++ b/sys/unix/Makefile.src
@@ -56,6 +56,9 @@ SYSSRC = ../sys/share/ioctl.c ../sys/share/unixtty.c ../sys/unix/unixmain.c \
 	../sys/unix/unixunix.c ../sys/unix/unixres.c
 SYSOBJ = $(TARGETPFX)ioctl.o $(TARGETPFX)unixmain.o $(TARGETPFX)unixtty.o \
 	$(TARGETPFX)unixunix.o $(TARGETPFX)unixres.o
+# JP
+JSRC = ../japanese/jconj.c ../japanese/jlib.c
+JOBJ = $(TARGETPFX)jconj.o $(TARGETPFX)jlib.o
 #
 # for Systos
 # SYSSRC = ../sys/atari/tos.c ../sys/share/pcmain.c ../sys/share/pcsys.c \
@@ -557,10 +560,10 @@ CHAINOBJ = $(TARGETPFX)wc_chainin.o $(TARGETPFX)wc_chainout.o \
 	$(TARGETPFX)wc_trace.o
 
 # .c files for this version (for date.h)
-VERSOURCES = $(HACKCSRC) $(HACKLIBSRC) $(SYSSRC) $(WINSRC) $(CHAINSRC) $(GENCSRC)
+VERSOURCES = $(HACKCSRC) $(HACKLIBSRC) $(SYSSRC) $(JSRC) $(WINSRC) $(CHAINSRC) $(GENCSRC)
 
 # .c files for all versions using this Makefile (for lint and tags)
-CSOURCES = $(HACKCSRC) $(HACKLIBSRC) $(SYSCSRC) $(WINCSRC) $(CHAINSRC) $(GENCSRC)
+CSOURCES = $(HACKCSRC) $(HACKLIBSRC) $(SYSCSRC) $(JSRC) $(WINCSRC) $(CHAINSRC) $(GENCSRC)
 
 
 # all .h files except date.h, which would
@@ -632,7 +635,7 @@ HOBJ = $(TARGETPFX)allmain.o $(TARGETPFX)alloc.o \
 	$(TARGETPFX)wield.o $(TARGETPFX)windows.o $(TARGETPFX)wizard.o \
 	$(TARGETPFX)wizcmds.o $(TARGETPFX)worm.o $(TARGETPFX)worn.o \
 	$(TARGETPFX)write.o $(TARGETPFX)zap.o \
-	$(REGEXOBJ) $(RANDOBJ) $(SYSOBJ) $(WINOBJ) \
+	$(REGEXOBJ) $(RANDOBJ) $(JOBJ) $(SYSOBJ) $(WINOBJ) \
 	$(HINTOBJ) $(SNDLIBOBJ) $(UUIDOBJ) \
 	$(TARGETPFX)version.o 
 
@@ -1310,6 +1313,10 @@ qt_xcmd.moc: $(QTn_H) ../win/Qt/qt_xcmd.h
 	$(MOCPATH) -o $@ ../win/Qt/qt_xcmd.h
 qt_yndlg.moc: $(QTn_H) ../win/Qt/qt_yndlg.h
 	$(MOCPATH) -o $@ ../win/Qt/qt_yndlg.h
+$(TARGETPFX)jconj.o: ../japanese/jconj.c $(HACK_H)
+	$(TARGET_CC) $(TARGET_CFLAGS) -c -o $@ ../japanese/jconj.c
+$(TARGETPFX)jlib.o: ../japanese/jlib.c $(HACK_H)
+	$(TARGET_CC) $(TARGET_CFLAGS) -c -o $@ ../japanese/jlib.c
 # DEPENDENCIES MUST END AT END OF FILE
 # IF YOU PUT STUFF HERE IT WILL GO AWAY
 # see make depend above
--- a/sys/unix/unixmain.c
+++ b/sys/unix/unixmain.c
@@ -65,6 +65,11 @@ main(int argc, char *argv[])
 
     early_init(argc, argv);
 
+#if 1 /*JP*/
+    /* iconvの初期化のために一度は呼び出す必要がある */
+    setkcode('U');
+#endif
+
 #if defined(__APPLE__)
     {
 /* special hack to change working directory to a resource fork when
